File: LivrusVirtus/FoxToolKit/FXPython/CadIdiomas.py
from FXPy.fox import *
from Cadastros import frmCadastros
from resource import Resource
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
import Principal

class frmCadIdiomas(frmCadastros):

  def __init__(self, clienteMDI, iconeMDI, menuMDI):  
    frmCadastros.__init__(self, clienteMDI, Resource.STR_CADIDIOMA, 
      iconeMDI, menuMDI, TEXTFIELD_INTEGER, True)

    self.Rotinas = RotinasGlobais.Rotinas
    self.ConsSQL = ConsultasSQL.ConsSQL

    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNNOVO, 
      frmCadIdiomas.onBtnNovo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNPESQUISAR, 
      frmCadIdiomas.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNANTERIOR, 
      frmCadIdiomas.onBtnAnterior)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNPROXIMO, 
      frmCadIdiomas.onBtnProximo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNSALVAR, 
      frmCadIdiomas.onBtnSalvar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNEXCLUIR, 
      frmCadIdiomas.onBtnExcluir)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNLIMPAR, 
      frmCadIdiomas.onBtnLimpar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadIdiomas.ID_FRMCADASTROS_BTNFECHAR, 
      frmCadIdiomas.onBtnFechar)
    FXMAPFUNC(self, SEL_CLOSE, 0, frmCadIdiomas.onClose)

    self.labDescricao.setText("Idioma: ")

  def ValidaDados(self, bTodosDados):
    if self.Rotinas.ValidaCampo(self.edtCodigo.getText(), Resource.STR_CODNINF):
   	  return False
    if bTodosDados:
  	  if self.Rotinas.ValidaCampo(self.edtDescricao.getText(), Resource.STR_IDININF):
    		return False        	
    return True
  
  def onBtnNovo(self, sender, sel, ptr):
    self.btnNovoClick()
    self.NovoDado(self.ConsSQL.Idioma('N', self.edtCodigo.getText(),""))	
    return True  
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(
      self.ConsSQL.Idioma('S', self.edtCodigo.getText(),""), Resource.STR_IDINENC)
    return True  
  
  def onBtnAnterior(self, sender, sel, ptr):
    self.registroAnterior()	
    return True  
  
  def onBtnProximo(self, sender, sel, ptr):
    self.registroProximo()
    return True  
  
  def onBtnSalvar(self, sender, sel, ptr):
    self.SalvarDados(self.ValidaDados(True),
      self.ConsSQL.Idioma('S', self.edtCodigo.getText(),""),
      self.ConsSQL.Idioma('U', self.edtCodigo.getText(), 
        self.edtDescricao.getText()),
      self.ConsSQL.Idioma('I', self.edtCodigo.getText(), 
        self.edtDescricao.getText()), 
      True)
    return True  
  
  def onBtnExcluir(self, sender, sel, ptr):
    self.ExcluirDados(self.ValidaDados(False),
      self.ConsSQL.Idioma('D', self.edtCodigo.getText(),""))
    return True  
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.btnLimparClick()
    return True  

  def onBtnFechar(self, sender, sel, ptr):
    self.onClose(sender, sel, ptr)
    return True
  
  def onClose(self, sender, sel, ptr):
    Principal.fPrincipal.habilitaMenu("CadIdiomas")
    self.destroy()
    return True

File: LivrusVirtus/FoxToolKit/FXRuby/CadIdiomas.rb
require 'fox'
require 'fox/responder'
require 'Cadastros'
require 'Principal'
require 'RotinasGlobais'
require 'ConsultasSQL'
include Fox
include Responder

class FRMCadIdiomas < FRMCadastros
  
  def initialize(clienteMDI, iconeMDI, menuMDI)  
    super(clienteMDI, Resource::STR_CADIDIOMA, iconeMDI, menuMDI, 
      TEXTFIELD_INTEGER, true)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNNOVO, :onBtnNovo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNANTERIOR, :onBtnAnterior)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPROXIMO, :onBtnProximo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNSALVAR, :onBtnSalvar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNEXCLUIR, :onBtnExcluir)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNFECHAR, :onBtnFechar)
    FXMAPFUNC(SEL_CLOSE, 0, :onClose)

    @labDescricao.setText("Idioma: ")
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(@edtCodigo.getText(), Resource::STR_CODNINF)
    	  return false
    end
  	if bTodosDados
  	  if $Rotinas.ValidaCampo(@edtDescricao.getText(), Resource::STR_IDININF)
    		return false
      end
  	end
  	return true
  end 
  
  def onBtnNovo(sender, sel, ptr)
    btnNovoClick()
    NovoDado($ConsSQL.Idioma('N', @edtCodigo.getText(),""))	
    return true
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    PesquisarDados(
      $ConsSQL.Idioma('S', @edtCodigo.getText(),""), Resource::STR_IDINENC)
    return true
  end
  
  def onBtnAnterior(sender, sel, ptr)
    registroAnterior()	
    return true
  end
  
  def onBtnProximo(sender, sel, ptr)
    registroProximo()
    return true
  end
  
  def onBtnSalvar(sender, sel, ptr)
    SalvarDados(ValidaDados(true),
      $ConsSQL.Idioma('S', @edtCodigo.getText(),""),
      $ConsSQL.Idioma('U', @edtCodigo.getText(), @edtDescricao.getText()),
      $ConsSQL.Idioma('I', @edtCodigo.getText(), @edtDescricao.getText()), 
      true)
    return true
  end
  
  def onBtnExcluir(sender, sel, ptr)
    ExcluirDados(ValidaDados(false),
      $ConsSQL.Idioma('D', @edtCodigo.getText(),""))
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    btnLimparClick()
    return true
  end

  def onBtnFechar(sender, sel, ptr)
    onClose(sender, sel, ptr)
    return true
  end

  def onClose(sender, sel, ptr)
    $frmPrincipal.habilitaMenu("CadIdiomas")
    destroy()
    return true
  end 
 
end
